### PR TITLE
add privacy notice and command to set update check false

### PR DESCRIPTION
**Pull request by tejal29**
Adding a config for turning off update check for skaffold

Test Notes:

1. Build a non dirty version with version set to < 0.37.0
```
GOOS=linux GOARCH=amd64 CGO_ENABLED=1 go install -tags "osusergo netgo static_build" -ldflags " -X github.com/GoogleContainerTools/skaffold/pkg/skaffold/version.version=v0.30.0 -X github.com/GoogleContainerTools/skaffold/pkg/skaffold/version.buildDate=2019-08-30T10:29:55Z -X github.com/GoogleContainerTools/skaffold/pkg/skaffold/version.gitCommit=13a5d595d7bcc760d00460e80dd78fa0f676d1ac -X github.com/GoogleContainerTools/skaffold/pkg/skaffold/version.gitTreeState=dirty  -extldflags \"-static\"" -gcflags "all=-trimpath=/usr/local/google/home/tejaldesai/workspace/skaffold1" -asmflags "all=-trimpath=/usr/local/google/home/tejaldesai/workspace/skaffold1" github.com/GoogleContainerTools/skaffold/cmd/skaffold
```

2. Verify skaffold version is < 0.37.0 and `GitTreeState` is non-dirty to by-pass this condition https://github.com/GoogleContainerTools/skaffold/blob/master///pkg/skaffold/update/update.go#L42
```
$ skaffold version -v=debug
INFO[0000] Skaffold &{Version:t0.40.0 ConfigVersion:skaffold/v1beta14 GitVersion: GitCommit:13a5d595d7bcc760d00460e80dd78fa0f676d1ac GitTreeState:test BuildDate:2019-08-30T10:29:55Z GoVersion:go1.12.5 Compiler:gc Platform:linux/amd64} 
v0.30.0
```

3. run skaffold deploy in examples/getting-started and verify we see the message for newer version
```
$ cd examples/geting-started
$ skaffold deploy
Starting deploy...
kubectl client version: 1.11+
kubectl version 1.12.0 or greater is recommended for use with Skaffold
 - pod/getting-started configured
Deploy complete in 949.223543ms
There is a new version (0.37.0) of Skaffold available. Download it at https://storage.googleapis.com/skaffold/releases/latest/skaffold-linux-amd64
```

4. Now disable the update-check config via command line 
```
$skaffold config set -g update-check false
set global value update-check to false
```

5. Run skaffold deploy again and make sure there is no message for newer skaffold version.
```
tejaldesai@@getting-started (add_privacy_policy)$ skaffold deploy 
Starting deploy...
kubectl client version: 1.11+
kubectl version 1.12.0 or greater is recommended for use with Skaffold
 - pod/getting-started configured
Deploy complete in 1.170797454s
```

You can test all scenarios in the unit test.

### CODE REVIEW

File: cmd/skaffold/app/cmd/cmd.go [hunk 1]
```diff
@@ -96,11 +96,13 @@ func NewSkaffoldCommand(out, err io.Writer) *cobra.Command {
 			logrus.Infof("Skaffold %+v", version)
 			event.LogSkaffoldMetadata(version)
 
+			SetUpFlags()
```
Comment (balopat) [inline cmd/skaffold/app/cmd/cmd.go]: should be `SetupFlags`

File: pkg/skaffold/update/update.go [hunk 2]
```diff
@@ -31,20 +32,32 @@ import (
 // Fot testing
 var (
 	GetLatestAndCurrentVersion = getLatestAndCurrentVersion
+	readConfig                 = config.ReadConfigFileNoCache
```
Comment (balopat) [inline pkg/skaffold/update/update.go]: ReadConfigFileNoCache is only for testing - use `ReadConfigFile` instead

File: pkg/skaffold/update/update.go [hunk 3]
```diff
@@ -31,20 +32,32 @@ import (
...
-	v := os.Getenv(constants.UpdateCheckEnvironmentVariable)
-	return v == "" || strings.ToLower(v) == "true"
+	return isUpdateCheckEnabledByEnvOrConfig(configfile)
+}
+
+func isUpdateCheckEnabledByEnvOrConfig(configfile string) bool {
+	if v := getEnv(constants.UpdateCheckEnvironmentVariable); v != "" {
+		return strings.ToLower(v) == "true"
+	}
+	cfg, err := readConfig(configfile)
```
Comment (balopat) [inline pkg/skaffold/update/update.go]: the config file reading logic should be in the config package with the other config parameters.

File: pkg/skaffold/update/update_test.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+		description string
+		envVariable string
+		cfg         *config.GlobalConfig
+		readErr     error
+		expected    bool
+	}{
+		{
+			description: "nothing is set opt-in behavior",
+			expected:    true,
+		}, {
```
Comment (balopat) [inline pkg/skaffold/update/update_test.go]: this testcase formatting is pretty but it's inconsistent from the other test cases: 
```suggestion
		},
		{
```

File: cmd/skaffold/app/cmd/cmd.go (hunk 1 above)
Comment (balopat) [reply cmd/skaffold/app/cmd/cmd.go]: I know it wasn't introduced here :)

File: pkg/skaffold/update/update.go (hunk 2 above)
Comment (tejal29) [reply pkg/skaffold/update/update.go]: I think there was a race detected via using ReadConfigFile. The reason being update check happens in background and both the background and foreground skaffold end up writing to the cache.
I will check again.

File: pkg/skaffold/update/update.go (hunk 3 above)
Comment (tejal29) [reply pkg/skaffold/update/update.go]: ah ok. i will move reading "UpdateCheck" to config.

File: pkg/skaffold/update/update.go (hunk 2 above)
Comment (tejal29) [reply pkg/skaffold/update/update.go]: This is no longer true. I think it was due to `SetUpFlags`.

**Comment by balopat** [comment]
Lgtm, I bounced the Travis integration job - it flaked out on an rules_docker download! You are unlucky with CI:)))

**Comment by tejal29** [comment]
Thanks @balopat on following up! Have a great weekend!

**Pull request merged**